### ISSUE TITLE
kubectl get pod <pod_name> should return the pod even if if it's terminated

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#41150]
**What happened**:

Pod `foo` failed. I want to inspect the status of the pod, but getting the pod directly returns nothing.
```console
$ kubectl get pod foo
$
```
Adding the `--show-all` (`-a`) option shows the pod.
```console
$ kubectl get pod foo -a
NAME      READY     STATUS    RESTARTS   AGE
foo       0/1       Error     0          12m
```

**What you expected to happen**:

I understand by default `get pod` only shows non-terminated resources, but since I specified the pod name explicitly, I'd expect to get the pod regardless of its state.

**Comment by AdoHe** [comment]
@kubernetes/sig-cli-misc

**Comment by xingzhou** [comment]
agree that this should be supported since the name of the pod is pointed out explicitly in the command.

**Comment by smarterclayton** [comment]
Yeah, filtering should not happen on get of one or more names.  It's only the list call that should filter.

**Comment by smarterclayton** [comment]
Setting to 1.6 because this is a usability issue.

**Comment by xingzhou** [comment]
I would like to work on this if nobody now is taking care of it.

**Comment by smarterclayton** [comment]
Fixed in #42216
